src/components/shared/Button.tsx

```diff
@@ -25,19 +28,30 @@ interface Props {
   isDisabled: boolean;
   containerStyle?: ViewStyle;
   style?: ViewStyle;
-  onPress: () => void;
+  onPress: (params?: any) => void | Promise<void>;
   imgLeftSrc?: ImageSourcePropType;
   indicatorColor: string;
   activeOpacity: number;
   children: string;
-  width: number;
-  height: number;
+  width?: ReactText;
+  height?: ReactText;
+  flexDirection?: string;
+  justifyContent?: string;
+  alignItems?: string;
 }
 
+const ButtonContainer = styled.View<ButtonContainer>`
+  flex: 1;
+  flex-direction: ${({ flexDirection }): string => flexDirection || 'row'};
+  ${({ justifyContent }): string | undefined => justifyContent && `justify-content: ${justifyContent}`};
```

Comment:
I don't think this is a good idea to pass down each `style prop` one by one. I think you should revert this. Instead try to pass down the whole style props for flexibility.